applications/HDF5Application/custom_io/hdf5_vertex_container_io.cpp

```diff
@@ -10,189 +10,122 @@
-    return parameters["prefix"].GetString();
-
-    KRATOS_CATCH("");
-}
-
-
-VertexContainerCoordinateIO::VertexContainerCoordinateIO(Parameters parameters,
-                                                         File::Pointer pFile)
-    : VertexContainerIO(VertexContainerCoordinateIO::FormatParameters(parameters), pFile)
+void VertexContainerCoordinateIO::Write(
+    const Detail::VertexContainerType& rVertices,
+    Parameters Attributes)
 {
-    KRATOS_TRY
-
-    parameters.AddMissingParameters(VertexContainerCoordinateIO::GetDefaultParameters());
-    mWriteIDs = parameters["write_vertex_ids"].GetBool();
-
-    KRATOS_CATCH("");
-}
-
-
-void VertexContainerCoordinateIO::Write(const Detail::VertexContainerType& rVertices)
-{
-    if (mWriteIDs)
-        WriteWithIDs(rVertices);
-    else
-        WriteWithoutIDs(rVertices);
+    if (mWriteIDs) {
+        WriteWithIDs(rVertices, Attributes);
+    } else {
+        WriteWithoutIDs(rVertices, Attributes);
+    }
 }
 
 
-void VertexContainerCoordinateIO::WriteWithIDs(const Detail::VertexContainerType& rVertices)
+void VertexContainerCoordinateIO::WriteWithIDs(
+    const Detail::VertexContainerType& rVertices,
+    Parameters Attributes)
 {
     KRATOS_TRY
 
-    const std::string path = mComponentPath + "/POSITION";
+    const std::string path = mPathPrefix + "/POSITION";
 
     // Collect vertex coordinates into a buffer
     Matrix<double> buffer;
-    buffer.resize(rVertices.size(),
-                  4,
-                  false);
+    buffer.resize(rVertices.size(), 4,  false);
 
-    for (std::size_t i_vertex=0; i_vertex<rVertices.size(); ++i_vertex) {
-        const auto& rVertex = *(rVertices.begin() + i_vertex);
+    IndexPartition<IndexType>(rVertices.size()).for_each([&rVertices, &buffer](const auto Index) {
+        const auto& rVertex = *(rVertices.begin() + Index);
 
-        for (std::size_t i_component=0; i_component<3; ++i_component) {
-            buffer(i_vertex, i_component) = rVertex[i_component];
+        for (std::size_t i_component=0; i_component < 3; ++i_component) {
+            buffer(Index, i_component) = rVertex[i_component];
         }
 
-        buffer(i_vertex, 3) = rVertex.GetID();
-    }
+        buffer(Index, 3) = rVertex.GetID();
+    });
 
     // Write buffer to the requested path
     WriteInfo write_info;
-    mpFile->WriteDataSet(path,
-                         buffer,
-                         write_info);
+    mpFile->WriteDataSet(path, buffer, write_info);
+    mpFile->WriteAttribute(path, Attributes);
 
     KRATOS_CATCH("");
 }
 
-
-void VertexContainerCoordinateIO::WriteWithoutIDs(const Detail::VertexContainerType& rVertices)
+void VertexContainerCoordinateIO::WriteWithoutIDs(
+    const Detail::VertexContainerType& rVertices,
+    Parameters Attributes)
 {
     KRATOS_TRY
 
-    const std::string path = mComponentPath + "/POSITION";
+    const std::string path = mPathPrefix + "/POSITION";
```

Comment:
Perhaps filesystem::path would be better here

Reply:
that wouldn't work on windows because it would translate to `...\\POSITION`, but this is a path within HDF5

Reply:
I think you can also get the path with /

Reply:
I am not sure, I went through the docs, but I am not sure whether they gurantee it.